ext/standard/streamsfuncs.c

```diff
@@ -1774,6 +1774,46 @@ PHP_FUNCTION(sapi_windows_vt100_support)
 		}
 	}
 }
+
+PHP_FUNCTION(sapi_windows_console_size)
+{
+	zval *zsrc;
+	php_stream *stream;
+	zend_long fileno;
+	int width, height;
+	zval zwidth, zheight;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+	Z_PARAM_RESOURCE(zsrc)
+	ZEND_PARSE_PARAMETERS_END();
+
+	php_stream_from_zval(stream, zsrc);
+
+	/* get the fd.
+	 * NB: Most other code will NOT use the PHP_STREAM_CAST_INTERNAL flag when casting.
+	 * It is only used here so that the buffered data warning is not displayed.
+	 */
+	if (php_stream_can_cast(stream, PHP_STREAM_AS_FD_FOR_SELECT | PHP_STREAM_CAST_INTERNAL) == SUCCESS) {
+		php_stream_cast(stream, PHP_STREAM_AS_FD_FOR_SELECT | PHP_STREAM_CAST_INTERNAL, (void*)&fileno, 0);
+	} else if (php_stream_can_cast(stream, PHP_STREAM_AS_FD | PHP_STREAM_CAST_INTERNAL) == SUCCESS) {
+		php_stream_cast(stream, PHP_STREAM_AS_FD | PHP_STREAM_CAST_INTERNAL, (void*)&fileno, 0);
```

Comment:
I had copied this code from `sapi_windows_vt100_support()` without thinking. Now I had a closer look, and wondered why we even try to cast to `PHP_STREAM_AS_FD_FOR_SELECT ` (we don't actually want to select). Looking at the implementation of `php_stdiop_cast()` it's clear: this cast won't flush (what we indeed don't want), but other possibly castable stream might not implement `PHP_STREAM_AS_FD_FOR_SELECT `. So these two cast attempts make some sense (although they rely on a leaky abstraction).

Then I wondered why we first check if we *can* cast and then actually *do* cast. Regarding the implementation that makes not much sense, but for the sake of abstraction it is correct. On the other hand, asking whether we can cast to `PHP_STREAM_AS_FD` actually flushes (for stdio stream), so this is somewhat (s/somewhat/completely) broken anyway.

Well, probably best to leave as is for now.